### PR TITLE
Add priorityClassName support

### LINKED ISSUE #321: [Feature request] Allow to set priorityClassName for master pod
There is a possibility to assign a priorityClass to pod and by this make pod either

- not be preempted by others
- preempt other pods, when there are not enough resources for it

See [Pod Priority and Preemption documentation](https://kubernetes.io/docs/concepts/configuration/pod-priority-preemption/) for more details.

It would be nice to have a possibility to set priorityClassName for JenkinsMaster, but I do not see one right now.

### PR DESCRIPTION
**Pull request by anxolerd**
## Description
Allow to set priorityClassName for master pod. This will make possible
for jenkins master to either not be preempted on other's pod low
resources or to preempt lesser priority pods on resources shortage.

## API changes
Add `spec.master.priorityClassName` field to `Jenkins` CR, which corresponds to `spec.priorityClassName` field in `Pod` spec. Field type is `string`, it should be either name of a valid `scheduling.k8s.io/v1/PriorityClass` or empty string.

## Linked issues
Closes #321

### CODE REVIEW

File: pkg/controller/jenkins/configuration/base/validate.go [hunk 1]
```diff
@@ -407,3 +414,17 @@ func (r *ReconcileJenkinsBaseConfiguration) validateCustomization(customization
 
 	return messages, nil
 }
+
+func (r *ReconcileJenkinsBaseConfiguration) validatePriorityClassName(priorityClassName string) ([]string, error) {
+	var messages []string
+	if len(priorityClassName) > 0 {
+		priorityClass := &schedulingv1.PriorityClass{}
+		err := r.Client.Get(context.TODO(), types.NamespacedName{Name: priorityClassName}, priorityClass)
```
Comment (tomaszsek) [inline pkg/controller/jenkins/configuration/base/validate.go:422]: One missing part. You have to update https://github.com/jenkinsci/kubernetes-operator/blob/master/deploy/role.yaml and give operator permission to get `PriorityClass`.

Comment (anxolerd) [reply pkg/controller/jenkins/configuration/base/validate.go:422]: Changed that in 88fa77d

**Comment by tomaszsek** [comment]
It will be great if we could check this in e2e https://github.com/jenkinsci/kubernetes-operator/blob/master/test/e2e/jenkins.go#L69.

**Comment by anxolerd** [comment]
@tomaszsek , I reverted validation, as it compicates things in multitenant environment and added e2e test.

I have a question regarding e2e tests: is it possible to check negative cases (i.e. when reconcile loop fails) in those?

**Comment by anxolerd** [comment]
Tests are failing due to this: https://github.com/kubernetes/kubernetes/issues/60596

**Comment by tomaszsek** [comment]
You can bump minikube Kubernetes version here https://github.com/jenkinsci/kubernetes-operator/blob/master/config.minikube.env#L3

**Comment by anxolerd** [comment]
@tomaszsek , after tons of struggling with e2e, i finally managed to make tests pass!

**Comment by tomaszsek** [comment]
@anxolerd Thanks :+1:

**Pull request merged**